### PR TITLE
[GEOT-5929][GEOT-5928] Upgrade Solr and fix integration tests

### PR DESCRIPTION
**Pull request by nmco**
Associated issues:
- https://osgeo-org.atlassian.net/browse/GEOT-5929
- https://osgeo-org.atlassian.net/browse/GEOT-5928

This tests upgrade SolrJ client to its last stable version (7.2.1), this version is compatible at least with the following apache Solr versions 4.x, 5.x, 6.x and 7.x.

This PR also fix the Apache Solr datastore online tests and provides documentation about how to run them. Note that the manual steps necessary to run the online were simplified.

### CODE REVIEW

File: modules/unsupported/solr/src/test/java/org/geotools/data/solr/TestsSolrUtils.java [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+     *  <field name="vendor_s">D-Link</field>
+     *  (...)
+     * </doc>
+     * }
+     * </pre>
+     *
+     * @param client Sorl client to use, should already point to the desired core
+     * @param xml    stream contain the Solr documents encoded in XML
+     */
+    public static void updateRequest(HttpSolrClient client, InputStream xml) {
```
Comment (aaime) [inline modules/unsupported/solr/src/test/java/org/geotools/data/solr/TestsSolrUtils.java]: Maybe "runUpdate". At first I thought the code was going to modify a request but saw nothing returned from the method...

File: docs/user/library/data/solr.rst [hunk 2]
```diff
@@ -39,4 +39,57 @@ Example use:
...
+
+        {
+          "responseHeader":{
+            "status":0,
+            "QTime":782},
+          "core":"test_core"}
+
+Note that the integration tests will be able to reuse the same core for multiple runs as long
+the core managed schema is not manually edited. If any error happens during the integration tests 
+execution it is strongly recommended to use a clean core for new runs.
```
[the remark this answers is not in the document]
Comment (nmco) [reply docs/user/library/data/solr.rst]: Good point, the core needs to be removed and recreated.

File: docs/user/library/data/solr.rst [hunk 3]
```diff
@@ -39,4 +39,57 @@ Example use:
         map.put("solr_url", new URL("http://localhost:8080/solr"));
         map.put(SolrDataStoreFactory.NAMESPACE.key, "namespace");
 
-       
+Online Tests
```
[the remark this answers is not in the document]
Comment (nmco) [reply docs/user/library/data/solr.rst]: Goo point, this instructions will be moved to a read file.

File: modules/unsupported/solr/src/test/java/org/geotools/data/solr/TestsSolrUtils.java (hunk 1 above)
Comment (nmco) [reply modules/unsupported/solr/src/test/java/org/geotools/data/solr/TestsSolrUtils.java]: I agree indeed the method name is misleading, I will also update all the other methods names that are actually running a request to contain the term run in its name.

**Comment by nmco** [comment]
Requested changes applied, thanks for the review @aaime !

**Comment by aaime** [comment]
+1, thanks

**Pull request merged**